stdlib/bytes.mli

```diff
@@ -329,6 +329,30 @@ val equal: t -> t -> bool
     always-correct {!Bytes.to_string} and {!Bytes.of_string} instead.
 *)
 
+val to_hex : bytes -> bytes
+(** Return the given byte sequence as a byte sequence of US-ASCII lowercase
+    hexadecimal digits.
```

Comment:
I find it strange to mention US-ASCII here. You could equally well write `sequence of UTF-8 lowercase hexadecimal digits`. You should mention both or maybe neither, since these two are all the reasonable encodings that exist.

Reply:
> You could equally well write `sequence of UTF-8 lowercase hexadecimal digits`

This would be confusing. 

First it's always better not to confuse code points and encodings, so something like `a sequence of UTF-8 encoded Unicode lowercase hexadecimal digits` should rather be written. But that would be, depending on your perspective, ambiguous or wrong.

Unicode has two properties to distinguish hexdecimal digits: [Hex_Digit](https://erratique.ch/software/uucp/doc/Uucp/Num/index.html#val-is_hex_digit) and [ASCII_Hex_Digit](https://erratique.ch/software/uucp/doc/Uucp/Num/index.html#val-is_ascii_hex_digit). They do not coincide and for example U+FF45 has both the `Lowercase` property and `Hex_Digit` property. This function however does not handle this character. 

I think the current wording is a precise and accurate description of the function and is consistent with other doc strings that work on US-ASCII (e.g. `ascii_lowercase`). The docs should rather educate people on the fact that US-ASCII code points are compatible with Unicode code points and that the US-ASCII encoding and `UTF-8` encoding of these code points are equal.